src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpSimplifyTypeNamesDiagnosticAnalyzer.cs

```diff
@@ -92,13 +92,7 @@ internal override bool IsCandidate(SyntaxNode node)
             else
             {
                 var expression = (ExpressionSyntax)node;
-
-                // in case of an As or Is expression we need to handle the binary expression, because it might be 
-                // required to add parenthesis around the expression. Adding the parenthesis is done in the CSharpNameSimplifier.Rewriter
-                var expressionToCheck = expression.Kind() == SyntaxKind.AsExpression || expression.Kind() == SyntaxKind.IsExpression
-                    ? ((BinaryExpressionSyntax)expression).Right
-                    : expression;
```

Comment:
so this condition can never happen.  Here's why:

This function (`CanSimplifyTypeNameExpression`) is only called from two location.  
1. from the simplification walker.  The walker only calls through from a set of well known types, none of which are `AsExpression/IsExpression`.
2. from the code fix provider for simplify type names.  However, this does a check first that it will only call into this code if `_analyzer.IsCandidate(node)` is true.  And `IsCandidate` only returns true for names/memberaccess/qualifiedcref.  Never for Is/As expressions.